patches/0004-Use-crypto-backends.patch

```diff
@@ -379,19 +380,162 @@ index 33942467784ad3..0282ffa9fa23c8 100644
  	"crypto/internal/cryptotest"
  	fipsaes "crypto/internal/fips140/aes"
  	"encoding/hex"
+diff --git a/src/crypto/cipher/gcm.go b/src/crypto/cipher/gcm.go
+index 5580f96d55a0fb..cc759af19b116b 100644
+--- a/src/crypto/cipher/gcm.go
++++ b/src/crypto/cipher/gcm.go
+@@ -93,7 +93,14 @@ func newGCM(cipher Block, nonceSize, tagSize int) (AEAD, error) {
+ func NewGCMWithRandomNonce(cipher Block) (AEAD, error) {
+ 	c, ok := cipher.(*aes.Block)
+ 	if !ok {
+-		return nil, errors.New("cipher: NewGCMWithRandomNonce requires aes.Block")
++		if cipher, ok := cipher.(gcmAble); ok {
++			g, err := cipher.NewGCM(gcmStandardNonceSize, gcmTagSize)
++			if err != nil {
++				return nil, err
++			}
++
++			return gcmWithRandomNonce{g}, nil
++		}
+ 	}
+ 	g, err := gcm.New(c, gcmStandardNonceSize, gcmTagSize)
+ 	if err != nil {
+@@ -103,7 +110,7 @@ func NewGCMWithRandomNonce(cipher Block) (AEAD, error) {
+ }
+ 
+ type gcmWithRandomNonce struct {
+-	*gcm.GCM
++	aead AEAD
+ }
+ 
+ func (g gcmWithRandomNonce) NonceSize() int {
+@@ -157,7 +164,12 @@ func (g gcmWithRandomNonce) Seal(dst, nonce, plaintext, additionalData []byte) [
+ 		plaintext = ciphertext[:len(plaintext)]
+ 	}
+ 
+-	gcm.SealWithRandomNonce(g.GCM, nonce, ciphertext, plaintext, additionalData)
++	if gc, ok := g.aead.(*gcm.GCM); ok {
++		gcm.SealWithRandomNonce(gc, nonce, ciphertext, plaintext, additionalData)
++	} else if gc, ok := g.aead.(randomNonceSealer); ok {
++		gc.SealWithRandomNonce(ciphertext, nonce, plaintext, additionalData)
++	}
```

Comment:
If you implement what I said in https://github.com/microsoft/go/pull/1673/files#r2108709392, then `g.aead` should always be either a `*gcm.GCM` or a `randomNonceSealer`. Therefore it is an error on our side if none of these cases are meet. I would add an else case that panics to be sure we haven't forgotten any case.